src/cargo/ops/cargo_check.rs

```diff
@@ -0,0 +1,89 @@
+use core::Workspace;
+use ops::{self, CompileOptions, MessageFormat};
+use util::important_paths::{find_root_manifest_for_wd};
+use util::{CliResult, Config};
+
+#[derive(RustcDecodable)]
+pub struct Options {
```

Comment:
Ah yeah the intention was that all the CLI options have their own suite of flags and such but everything "compiles down" to [`CompileOptions`](https://www.openssl.org/docs/manmaster/man3/SSL_write_ex.html) where that's the common functionality amongst everything.